src/index.mjs

```diff
@@ -175,3 +223,46 @@ export function prefetch(url, isPriority, conn) {
       })
   );
 }
+
+/**
+* Prerender a given URL
+* @param {String} url - the URL to fetch
+* @param {Object} [conn] - navigator.connection (internal)
+* @return {Object} a Promise
+*/
+export function prerender(urls, conn) {
+  let chkConn = checkConnection(conn = navigator.connection);
+  if(chkConn instanceof Error) {
+    return Promise.reject(new Error('Cannot prerender, '+chkConn.message));
+  }
+  
+  // prerendering preconditions:
+  // 1) whether UA supports spec rules.. If not, fallback to prefetch
+  if (!hasSpecRulesSupport()) {
+    prefetch (urls);
+    return Promise.reject(new Error('This Browser does not support Speculation Rules API. Falling back to Prefetch.'));
+  }
+  
+  // 2) whether spec rules is already defined (and with this we also covered when we have created spec rules before)
+  if (isSpecRulesExists()) {
+    return Promise.reject(new Error('Speculation Rules is already defined and cannot be altered.'));
+  }
+    
+  // 3) whether it's a same origin url,
+  try{
+    [].concat(urls).filter(isSameOrigin).map(str => {
+      if (!toPrerender.has(str)) {
+        toPrerender.add(str);
+      }
+    });
+  } catch (e) {
+    return Promise.reject(new Error('Only same origin URL(s) are allowed: '+e));
+  }
```

Comment:
```suggestion
  // 3) whether it's a same origin url,
  for (const url of [].concat(urls)) {
    if (!isSameOrigin(url)) {
      return Promise.reject(new Error('Only same origin URLs are allowed: ' + url));
    }

    toPrerender.add(url);
  }
```

This change relies on my other change to make `isSameOrigin` return true or false.